src/main/scala/beam/agentsim/agents/BeamAgent.scala

```diff
@@ -83,23 +83,23 @@ trait BeamAgent[T] extends LoggingFSM[BeamAgentState, T] with Stash {
       case None =>
         ""
     }
-    s"$tickStr${triggerStr}State:$stateName ${logPrefix()}$msg"
+    s"$tickStr${triggerStr}State:$stateName ${logPrefix()}"
   }
 
-  def logInfo(msg: String): Unit = {
-    log.info(s"${logWithFullPrefix(msg)}")
+  def logInfo(msg: => String): Unit = {
+    log.info("{} {}", getPrefix, msg)
   }
 
-  def logWarn(msg: String): Unit = {
-    log.warning(s"${logWithFullPrefix(msg)}")
+  def logWarn(msg:  => String): Unit = {
+    log.warning("{} {}", getPrefix, msg)
   }
 
-  def logError(msg: String): Unit = {
-    log.error(s"${logWithFullPrefix(msg)}")
+  def logError(msg:  => String): Unit = {
+    log.error("{} {}", getPrefix, msg)
   }
 
   def logDebug(msg: String): Unit = {
-    log.debug(s"${logWithFullPrefix(msg)}")
+    log.debug("{} {}", getPrefix, msg)
```

Comment:
You missed turning this one lazy. Also, I noticed you opted to not make this take args itself. Just as an FYI I'm not sure how string interpolation works with lazy evaluation from an allocation timing perspective.

Reply:
Generated code also proves that it's lazy:
![image](https://user-images.githubusercontent.com/5107562/44596354-fa1f9500-a7f5-11e8-8cef-386e85a268d8.png)